.github/actions/build_deb.sh

```diff
@@ -35,51 +39,100 @@ if [ -n "${CI_RUN+1}" ] ; then
     echo "silkit_debian_revision=${SILKIT_DEBIAN_REVISION}" >> "$GITHUB_OUTPUT"
 fi
 
+if [ -d ${SILKIT_SOURCE_URL} ] ; then
+    echo "SILKIT-PKG: Using local checkout!"
+    silkit_path="${SILKIT_SOURCE_URL}"
 
-if [ -n SILKIT_REVISION ] ; then
-    CLONE_VERSION="main"
 else
-    CLONE_VERSION="sil-kit/v${SILKIT_VERSION}"
-fi
+    ## Set GIT Options
+    if [ -n SILKIT_VENDORED_PACKAGES ] ; then
+        echo "Vendoring Packages!"
+        SUBMODULE_CMD="--recurse-submodules --shallow-submodules"
+    else
+        SUBMODULE_CMD=""
+    fi
 
-git -c http.sslVerify=false clone ${SILKIT_SOURCE_URL} -b ${CLONE_VERSION} libsilkit-${SILKIT_VERSION}
-ret_val=$?
-if [ "$ret_val" != '0' ] ; then
-    echo "SILKIT-PKG: Could get SIL Kit sources at ${SILKIT_SOURCE_URL}:${SILKIT_REVISION}"
-    echo "Trying local directory at ${PWD}/libsilkit-${SILKIT_VERSION}"
-fi
+    # Since we want to allow building SIL KIT from VERSIONS different from the current CHANGELOG Version
+    # We need to distinguish how to checkout and reset the local git repo
+    if [ -n SILKIT_REVISION ] ; then
+        CLONE_VERSION="main"
+    else
+        CLONE_VERSION="sil-kit/v${SILKIT_VERSION}"
```

Comment:
The `if` above tests a variable named `SILKIT_REVISION`, but it uses `SILKIT_VERSION` - i this correct?

Reply:
Ah I see `SILKIT_REVISION` is the git ref.

Reply:
Yes if we specify the git ref, we just clone main and then reset to the correct ref!